Fix ip header id field sequence in generated packets

Pavel4e5 opened the pull request:
Если задана последовательность фейковых пакетов, то в текущем виде все они имеют одинаковое поле Identification в заголовке IP пакета, это сказывается на проходимости DPI такими пакетами. Например для доступа к _*.googlevideo.com_ (Ростелеком) у меня достаточно в Zapret задать: `winws.exe --wf-tcp=443 --dpi-desync=fake --dpi-desync-fooling=badseq --dpi-desync-fake-tls=0x00 --dpi-desync-repeats=20` .
В GoodbyeDPI, в текущей версии программы, аналогичная последовательность  не работает - `goodbyedpi.exe --wrong-seq --fake-from-hex 00 --fake-resend 20` из за того, что вышеуказанное поле всегда имеет одно и то же значение.
Предлагаемый PR исправляет этот момент, и значение в этом поле увеличивается на единицу с каждым сгенерированным пакетом.

If a sequence of fake packets is given, in its current form they all have the same Identification field in the IP packet header, which affects the ability of such packets to pass through DPI. For example, to access *.googlevideo.com (Rostelecom), it is enough for me to set in Zapret: `winws.exe --wf-tcp=443 --dpi-desync=fake --dpi-desync-fooling=badseq --dpi-desync-fake-tls=0x00 --dpi-desync-repeats=20`.
In GoodbyeDPI, a similar sequence in the current version of the program does not work: `goodbyedpi.exe --wrong-seq --fake-from-hex 00 --fake-resend 20` because the aforementioned field always has the same value.
The proposed PR fixes this issue by incrementing value in this field by one, with each generated packet.

Pavel4e5 committed:
Fix ip header id field sequence

Pavel4e5 committed:
Merge branch 'master' into ip-header-id-field

MrCryptor commented:
Почему так долго идёт процесс одобрения этого фикса?

Pavel4e5 commented:
Да, тут долго рассматривают) . Если нужны бинарники с фиксом, то можно пока скачать у меня в Actions: [https://github.com/Pavel4e5/GoodbyeDPI/actions/runs/11728180799](https://github.com/Pavel4e5/GoodbyeDPI/actions/runs/11728180799)

Pavel4e5 commented:
>  По какой причине было выбрано случайное значение?

1. Минимальные изменения в коде.
2. Достигает результата.
3. По окончании добавленных пакетов последовательность поля id в любом случае ломается.

Если необходимо, могу изменить, чтобы копировать из предыдущего пакета (ACK) и увеличивать.

albamef commented:
> Да, тут долго рассматривают) . Если нужны бинарники с фиксом, то можно пока скачать у меня в Actions: https://github.com/Pavel4e5/GoodbyeDPI/actions/runs/11728180799

А можете перевыложить скомпилированный проект, пожалуйста?

Pavel4e5 commented:
> А можете перевыложить скомпилированный проект, пожалуйста?

https://github.com/Pavel4e5/GoodbyeDPI/actions/runs/14195558714

Pavel4e5 committed:
fix build.yml

Removed auto-update for man-db to prevent conflicts during MinGW-w64 installation.
